### ISSUE TITLE
Rotate180 may be hard-coded for 24-bit 

### ISSUE THREAD

**Issue by fire-eggs** [rageworx/fl_imgtk#14]
I've been having some problems with rotate180. 

1. Images of depth 4 (e.g. PNG with transparency), the transparent bits aren't rotated properly.
2. Images of depth 1 seem to result in a memory corruption crash when displayed.

I believe these lines:
```
            fl_imgtk_swap_mem( &buff[ cnt * d ],
                               &buff[ (imgmax - cnt - 1) * d ],
                               3);
```
should be:
```
            fl_imgtk_swap_mem( &buff[ cnt * d ],
                               &buff[ (imgmax - cnt - 1) * d ],
                               d);
```
I.e. not hard-coding 24 bits of image depth.

Here is a depth 1 PNG file:
[3016021.png.zip](https://github.com/rageworx/fl_imgtk/files/6489692/3016021.png.zip)

And here is a depth 4 PNG:
[19000.png.zip](https://github.com/rageworx/fl_imgtk/files/6489698/19000.png.zip)

**Comment by rageworx** [comment]
Dear Kevin,
Thank you for your advice, it must be my fault ( maybe it belong to merge from old source ).
I will just fix it now.

**Comment by rageworx** [comment]
Latest change tested with 32bit PNG:
![image](https://user-images.githubusercontent.com/6948225/118431425-4c751280-b711-11eb-950f-7f6dfd581d27.png)
You can refer to these prebuilt win64pe image zip:
[fltk_fl_imgtk_test_gee_techdemo_win64.zip](https://github.com/rageworx/fl_imgtk/files/6491062/fltk_fl_imgtk_test_gee_techdemo_win64.zip)

**Comment by rageworx** [comment]
Commited this issue for 1cdf2bf..272e2ff.
You can pull, and test it agian.
FL_IMGTK version build may increased to 26 from 25.

**Comment by rageworx** [comment]
Update : fixed some RGBA source issues,  commit of 272e2ff..528aa48.
Please check latest changes.

**Comment by fire-eggs** [comment]
Looks good to me, I'll try out the other changes soon!
